GlotPress/GlotPress-WP
# Fix issue with plugin activation through wp-cli, when database tables don't exist

## Problem
The plugin cannot be activated through wp-cli when the database tables don't exist yet.

```shell
# on a fresh clone of the repo

$ npm run env:start
$ wp-env run cli wp plugin deactivate GlotPress
$ wp-env run cli wp plugin activate GlotPress

WordPress database error Table 'wordpress.wp_gp_permissions' doesn't exist for query
SELECT * FROM wp_gp_permissions WHERE action = 'admin'
made by
include('phar:///usr/local/bin/wp/php/boot-phar.php'),
include('phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/wp-cli.php'),
WP_CLI\bootstrap,
WP_CLI\Bootstrap\LaunchRunner->process,
WP_CLI\Runner->start, WP_CLI\Runner->run_command_and_exit,
WP_CLI\Runner->run_command,
WP_CLI\Dispatcher\Subcommand->invoke,
call_user_func,
WP_CLI\Dispatcher\CommandFactory::WP_CLI\Dispatcher\{closure},
call_user_func,
Plugin_Command->activate, activate_plugin,
do_action('activate_GlotPress/glotpress.php'),
WP_Hook->do_action,
WP_Hook->apply_filters,
gp_activate_plugin,
GP_Thing->find_one,
GP_Thing->one
```

## Solution
The cause of this issue is that the code to upgrade the database only runs when a user visits an admin page. However, the plugin should be functional regardless of whether the user visited an admin page or not. So in this PR we run the database upgrade code on plugin activation.

I've also did a very minor refactor, to move database upgrade code out of `gp-settings.php` and into `glotpress.php`, please see each commit's message for details.

## To-do

- [x] Move database upgrade code out of `gp-settings.php` and into `glotpress.php`
- [x] Run database upgrade logic on plugin activation

## Testing Instructions

1. Clone the repo
2. `npm run env:start`
3. `wp-env run cli wp plugin deactivate GlotPress`
4. `wp-env run cli wp plugin activate GlotPress`
5. There should be no errors in the output of the command
